variables.tf

```diff
@@ -3,139 +3,169 @@
 # ------------------------------------------------------------------------------
 
 variable "apply_immediately" {
-  type        = "string"
   description = "Specifies whether any database modifications are applied immediately, or during the next maintenance window"
+  type        = string
   default     = "false"
 }
 
 variable "name_prefix" {
   description = "A prefix used for naming resources."
+  type        = string
+  default     = null
 }
 
 variable "vpc_id" {
   description = "The VPC ID."
+  type        = string
+  default     = null
 }
 
 variable "subnet_ids" {
   description = "A list of VPC subnet IDs."
-  type        = "list"
+  type        = list(string)
+  default     = null
```

Comment:
Not optional 😅 In general, I don't think we should add default values where there previous were not default values, at least not as part of the `0.12` upgrade (we can do it in a separate PR if required)

Reply:
Yes, but there is plenty of things we can just leave out and work with them the same way we did before. For example, all outputs/inputs could also be written as "${var.variable}" and tf12 accepts it anyway, also, in many cases, types do not need to be strictly defined as well.

It depends, what would you prefer?